src/linear_algebra/matrix.lean

```diff
@@ -936,3 +936,43 @@ def alg_equiv_matrix {R : Type v} {M : Type w} {n : Type*} [fintype n]
   [comm_ring R] [add_comm_group M] [module R M] [decidable_eq n] {b : n → M} (h : is_basis R b) :
   module.End R M ≃ₐ[R] matrix n n R :=
 h.equiv_fun.alg_conj.trans alg_equiv_matrix'
+
+/-- `std_basis_vec` is the standard basis for the free module `n → R`. -/
+def std_basis_vec {R : Type v} [ring R] {n : Type w} [decidable_eq n]
+  (i : n) : n → R := linear_map.std_basis R (λ _, R) i 1
+
+namespace std_basis_vec
+
+open_locale classical
+
+variables {R : Type v} [ring R] {n : Type w}
+
+lemma neq_eq_zero {i j : n} (h : i ≠ j) : std_basis_vec i j = (0 : R) :=
+by { rw [std_basis_vec, linear_map.std_basis_ne], exact h.symm }
+
+lemma eq_eq_one {i j : n} (h : i = j) : std_basis_vec i j = (1 : R) :=
+by { convert linear_map.std_basis_same _ (λ _, R) j 1 }
+
```

Comment:
Please add a version `eq_one` where you only use `i` (and substitute `i` for `j`). That can then be marked as a `@[simp`-lemma.